linux/Jenkinsfile

```diff
@@ -1,126 +1,134 @@
-NODE_LABEL = "dockerBuild&&linux&&x64" // Dynamic docker machines from Azure
+/*
+--------------------------------------------
+buildArch   RPM         DEB         Comments
+--------------------------------------------
+x86_64      x86_64      amd64
+armv7l      armv7hl     armhf       arm32
+aarch64     aarch64     arm64       arm64
+ppc64le     ppc64le     ppc64le
+source      src         -           only for SRPM and no need specify as option or target
+s390x       s390x       s390x       only for jdk8+
+*/
+
+env.NODE_LABEL_RPM = "dockerBuild&&linux&&x64" // RedHat + Suse + Debian x64
+env.NODE_LABEL_DEB = "docker&&linux&&${ARCH}"  // Debian non-x64 
+env.PRODUCT = "temurin"
 
 pipeline {
     agent {
-        label NODE_LABEL
+        label NODE_LABEL_RPM
+    }
+    options {
+        timeout(time: 2, unit: 'HOURS') 
     }
     parameters {
-        string(name: 'PRODUCT', defaultValue: 'temurin')
-        string(name: 'JDK_SOURCE', defaultValue: 'Adoptium')
-        choice(name: 'VERSION_FILTER', choices: ['8', '11', '17', '18', 'all'], description: 'Build for specific VERSION')
-        choice(name: 'DISTRO_FILTER', choices: ['all', 'Debian', 'RedHat', 'Suse'], description: 'Build for specific Distro')
-        booleanParam(name: 'UPLOAD', defaultValue: false, description: 'Tick this box to upload the deb/rpm files to Artifactory')
+        choice(name: 'VERSION', choices: ['8', '11', '17', '18'], description: 'Build for specific JDK VERSION')
+        choice(name: 'ARCH', choices: ['x86_64', 'armv7l', 'aarch64', 'ppc64le', 's390x', 'all'], description: "Build for specific platform\n s300x not for VERSION 8\n")
+        choice(name: 'DISTRO', choices: ['all', 'Debian', 'RedHat', 'Suse'], description: 'Build for specific Distro\n Select RPM builds for RedHat and Suse')
+        booleanParam(name: 'uploadPackage', defaultValue: false, description: 'Tick this box to upload the deb/rpm files exlude src.rpm to Artifactory')
+        booleanParam(name: 'uploadSRCRPM', defaultValue: false, description: 'Tick this box to upload the src.rpm files to Artifactory')
+        string(name: 'gitrepo', defaultValue: 'https://github.com/adoptium/installer', description: 'Do not modify unless want to build from a specific gitrepo')
+        string(name: 'gitbranch', defaultValue: 'master', description: 'Do not change unless want to build on a specific branch from above gitrepo')
+        booleanParam(name: 'enableDebug', defaultValue: false, description: 'Tick to enable --stacktrace for gradle build')
+    }
+    tools {
+        // this is set in the jenkins global tool
+        jdk "jdk-11.0.13+8"
```

Comment:
Should try to get this to 11.0.16.1

Reply:
if 11.0.16.1 is available on ci.adoptopenjdk, i can uplift the version

Reply:
![Screenshot from 2022-09-08 11-51-21](https://user-images.githubusercontent.com/915053/189092251-73ee623c-f550-4fb2-b332-bd95547900c6.png)
only see 11.0.13+8

Reply:
did test we need 11.0.16.1 installed as global "tools" before we can set to use it
```
WorkflowScript: 14: Tool type "jdk" does not have an install of "11.0.16.1" configured - did you mean "jdk-11.0.13+8"? @ line 14, column 13.
           jdk "11.0.16.1"
```

Reply:
We can make this change later